### DISCUSSION TITLE
Hiding php extension in requests

### DISCUSSION THREAD

**Discussion by gl2007** [codeigniter4/shield#814] [Q&A] [answered]
I integrated CI Shield into my app. Now I notice that the login and other requests use .php at the end, which I want to customize to hide them. (I am handling it via .htaccess file.) How do I achieve hiding .php for all requests generated by CI Shield?

**Comment by datamweb** [comment]
Hello, this issue is not directly related to Shield. Try through below. Otherwise, ask your question in the [forum](https://forum.codeigniter.com/).
```
RewriteEngine On
RewriteCond %{REQUEST_FILENAME} !-f
RewriteRule ^([^\.]+)$ $1.php [NC,L]
```

**Reply by gl2007** [reply]
@datamweb not sure I follow. If I am not signed in, it redirects me to **index.php/login**. Are you saying that this "index.php" is generated by CI and not Shield? In the **app\Config\Auth.php**, I see these 

```php
public array $views = [
        'login'                       => '\CodeIgniter\Shield\Views\login',
        'register'                   => '\CodeIgniter\Shield\Views\register',
        'layout'                     => '\CodeIgniter\Shield\Views\layout',
      ]; 
```
which are views, but I don't see how these requests are generated as I don't see appropriate configurations anywhere, and hence I am assuming Shield generates those requests and not CI directly. 

I have this by default from the CI app starter kit:

	RewriteCond %{REQUEST_FILENAME} !-f
	RewriteCond %{REQUEST_FILENAME} !-d
	RewriteRule ^([\s\S]*)$ index.php/$1 [L,NC,QSA]

**Reply by datamweb** [reply]
Do you want **damin.com/login** to be shown instead of **damin.com/index.php/login**?

**Reply by gl2007** [reply]
I just want .php to be removed or flexibility to remove it based on some configuration. So it can be domain.com/index/login

**Reply by datamweb** [reply, marked as answer]
>  So it can be domain.com/index/login

The domain.com/index/login causes problems, because index is recognized as a controller and the login method. It is better to do the following to remove it altogether.

Try set as follows in **app/Config/App.php**:

```diff
-public string $indexPage = 'index.php';
+public string $indexPage = '';
```
https://github.com/codeigniter4/CodeIgniter4/blob/127ce6b7a0ff7e0d9050837c031fb417741f1eb3/app/Config/App.php#L44

Probably, if you raise the issue in the forum, they will be able to help

**Reply by gl2007** [reply]
Thank you!! Yeah, I observed the exact things, i.e. index/login causing problems, and was wondering where I will route the index/login to. So I went ahead and removed it altogether. Tested all scenarios and it is working great!

**Reply by kenjis** [reply]
I sent a PR to improve the docs.
https://github.com/codeigniter4/CodeIgniter4/pull/7873
Can someone review it?
